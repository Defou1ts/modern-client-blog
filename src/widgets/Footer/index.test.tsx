import React from 'react';

import { render, screen } from '@testing-library/react';
import { useRouter } from 'next/router';

import renderer from 'react-test-renderer';

import { Footer } from './index';

import type { NextRouter } from 'next/router';

jest.mock('next/router', () => ({
	useRouter: jest.fn(),
}));

describe('Footer', () => {
	beforeAll(() => {
		const useRouterMock = {
			locale: 'en',
			locales: ['en', 'ru'],
			pathname: '/',
		};
		(useRouter as jest.Mock<NextRouter>).mockReturnValue(useRouterMock as NextRouter);
	});

	it('renders a Footer', () => {
		render(<Footer />);

		const titleText = screen.getByText('Modsen Client Blog');
		expect(titleText).toBeInTheDocument();
		expect(titleText).toBeVisible();
	});

	it('button header snapshot', () => {
		const component = renderer.create(<Footer />);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
