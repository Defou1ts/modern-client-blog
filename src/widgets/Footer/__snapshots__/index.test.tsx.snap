// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer button header snapshot 1`] = `
<footer
  className="footer"
>
  <div
    className="navbar"
  >
    <h1
      className="title"
    >
      Modsen Client Blog
    </h1>
    <nav
      className="nav"
    >
      <a
        className="navlink"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        header.home
      </a>
      <a
        className="navlink"
        href="/blog"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        header.blog
      </a>
      <a
        className="navlink"
        href="/contact"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        header.about
      </a>
      <a
        className="navlink"
        href="/about"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        header.contact
      </a>
      <a
        className="navlink"
        href="/privacy"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        header.privacy
      </a>
    </nav>
  </div>
</footer>
`;
